### DISCUSSION TITLE
devops swagger

### DISCUSSION THREAD

**Discussion by neottil** [eclipse-ditto/ditto#1621] [Q&A]
In documentation there is a section that describe devops http api: https://www.eclipse.org/ditto/installation-operating.html#retrieve-all-log-levels.
Is there a swagger file for this, like other api exposed by swagger module? 
Like this: https://github.com/eclipse-ditto/ditto/blob/master/documentation/src/main/resources/openapi/ditto-api-2.yml

**Comment by thjaeckle** [comment]
No, there currently is no swagger documentation for the DevOps APIs.

**Comment by neottil** [comment]
If it could be useful, I could try to make this contribution. I need guidance on  where to put it.

**Reply by thjaeckle** [reply]
If it is of value to you, it probably is also for others.
The sources of the OpenAPI docs are here:
https://github.com/eclipse-ditto/ditto/tree/master/documentation/src/main/resources/openapi/sources

There is e.g. the [api-2-index.yml](https://github.com/eclipse-ditto/ditto/blob/master/documentation/src/main/resources/openapi/sources/api-2-index.yml) file which contains references to all "sub-apis" and schemas - located in subfolders.
There is a build-process using npm in place to generate the [ditto-api-2.yml](https://github.com/eclipse-ditto/ditto/blob/master/documentation/src/main/resources/openapi/ditto-api-2.yml) (please do not edit this file manually, but generate it using the documented way in [README.md](https://github.com/eclipse-ditto/ditto/blob/master/documentation/src/main/resources/openapi/README.md).

**Reply by neottil** [reply]
Ok, I'll open an issue right away and try to work on it.
